app/src/main/kotlin/de/digitalService/useID/util/AbTestManager.kt

```diff
@@ -0,0 +1,80 @@
+package de.digitalService.useID.util
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import de.digitalService.useID.analytics.IssueTrackerManagerType
+import de.digitalService.useID.analytics.TrackerManagerType
+import de.digitalService.useID.hilt.ConfigModule
+import io.getunleash.UnleashClient
+import io.getunleash.UnleashConfig
+import io.getunleash.UnleashContext
+import io.getunleash.polling.PollingModes
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.coroutines.resume
+
+@Singleton
+class AbTestManager @Inject constructor(
+    @Named(ConfigModule.UNLEASH_API_URL) private val url: String,
+    @Named(ConfigModule.UNLEASH_API_KEY) private val apiKey: String,
+    private val trackerManager: TrackerManagerType,
+    private val issueTrackerManager: IssueTrackerManagerType
+) {
+
+    private lateinit var unleashClient: UnleashClient
+
+    private val _isSetupIntroTestVariant = mutableStateOf(false)
+    val isSetupIntroTestVariant: State<Boolean> = _isSetupIntroTestVariant
+
+    suspend fun initialise() = suspendCancellableCoroutine { continuation ->
+        val config = UnleashConfig.newBuilder()
+            .proxyUrl(url)
+            .clientKey(apiKey)
+            .pollingMode(PollingModes.autoPoll(Int.MAX_VALUE.toLong()))
+            .build()
+
+        val supportedAbTests = AbTest.values().joinToString(",") { it.testName }
+        val context = UnleashContext.newBuilder()
+            .appName("useid-android")
+            .sessionId(UUID.randomUUID().toString())
+            .properties(mutableMapOf(Pair("supportedToggles", supportedAbTests)))
+            .build()
+
+        unleashClient = UnleashClient(config, context)
+
+        unleashClient.addTogglesUpdatedListener {
+            _isSetupIntroTestVariant.value = isVariationActivatedFor(AbTest.SETUP_INTRODUCTION_EXPLANATION)
+            continuation.resume(Unit)
+        }
+        unleashClient.addTogglesErroredListener {
+            issueTrackerManager.capture(it)
+            trackDisabled("Error fetching toggles.")
+            continuation.resume(Unit)
+        }
+
+        continuation.invokeOnCancellation {
+            unleashClient.close()
+            trackDisabled("Timed out while fetching toggles.")
+        }
+    }
+
+    private fun isVariationActivatedFor(test: AbTest): Boolean =
+        if (unleashClient.isEnabled(test.testName)) {
+            val variantName = unleashClient.getVariant(test.testName).name
+            trackerManager.trackEvent("abtesting", test.testName, variantName)
+            variantName == "variation"
+        } else {
+            false
+        }
+
+    private fun trackDisabled(message: String) {
+        issueTrackerManager.addInfoBreadcrumbs("Unleash", message)
```

Comment:
After talking to Andreas we actually switched from breadcrumbs for issues with Unleash to errors. 
The commit is here: https://github.com/digitalservicebund/useid-app-ios/pull/177/commits/29b30bc70ac309c53c393a772ac18c5008b4ce65
(Doesn’t mean you need to do the same, but our reasoning was that breadcrumbs are only part of the error’s log and when no error is sent, we won’t get this data). We added breadcrumbs when value is requested though, so for other errors we can see when a user was in a test in case it helps with investigation (line 67).